### PR TITLE
Change the following storage fields from 256 bits to 128 bits

### PR DESCRIPTION
**Pull request by barakman**
The goal of this change is to [reduce trading gas cost](https://docs.google.com/spreadsheets/d/1Vhrl31SsRJKWBxhKdKCGtZIEZVoGMVWmfd1cteyGON8/edit#gid=966433680).

---

Change the following storage fields from 256 bits to 128 bits:
- `PoolLiquidity.bntTradingLiquidity`
- `PoolLiquidity.baseTokenTradingLiquidity`

---

Please review this under the notion that it is an initial draft of the required changes, designated for deciding whether or not we want to proceed in the suggested direction at this point.

### CODE REVIEW

File: contracts/pools/PoolMigrator.sol [hunk 1]
```diff
@@ -152,8 +154,8 @@ contract PoolMigrator is IPoolMigrator, Upgradeable, Utils {
             averageRate: data.averageRate,
             depositLimit: data.depositLimit,
             liquidity: PoolLiquidity({
-                bntTradingLiquidity: data.liquidity.bntTradingLiquidity,
-                baseTokenTradingLiquidity: data.liquidity.baseTokenTradingLiquidity,
+                bntTradingLiquidity: SafeCast.toUint128(data.liquidity.bntTradingLiquidity),
```
Comment (yudilevi) [inline contracts/pools/PoolMigrator.sol:157]: pretty sure both values are uint128 here

**Review by yudilevi — CHANGES_REQUESTED** [review]

File: contracts/pools/PoolMigrator.sol (hunk 1 above)
Comment (barakman) [reply contracts/pools/PoolMigrator.sol:157]: They are in `struct PoolLiquidity`.
However, here, the type of variable `data` is `struct PoolLiquidityV1`.

File: contracts/pools/PoolCollection.sol [hunk 2]
```diff
@@ -946,8 +946,8 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
             amounts.poolTokenTotalSupply
         );
 
-        liquidity.baseTokenTradingLiquidity = amounts.newBaseTokenTradingLiquidity;
-        liquidity.bntTradingLiquidity = amounts.newBNTTradingLiquidity;
+        liquidity.baseTokenTradingLiquidity = toUint128(amounts.newBaseTokenTradingLiquidity);
```
[the remark this answers is not in the document]
Comment (barakman) [reply contracts/pools/PoolCollection.sol]: done

File: contracts/pools/PoolMigrator.sol (hunk 1 above)
Comment (barakman) [reply contracts/pools/PoolMigrator.sol]: fixed

**Review by yudilevi — APPROVED** [review]

**Review by lbeder — APPROVED** [review]

**Pull request merged**